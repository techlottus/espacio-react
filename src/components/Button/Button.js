

import React, { useEffect } from 'react'


export const Button = React.memo(({ data, onClick }) => {



  const buttonRef = React.createRef();
  
  useEffect(() => {
    buttonRef.current.data = {
      id: data.id || null,
      type: data.type  || 'primary',
      title: data.title || '',
      size: data.size || 'small',
      icon: data.icon || '',
      lyIcon: data.lyIcon,
      disabled: data.disabled,
      isExpand: data.isExpand,
      test: data.test || ''
    }
  }, [data]) // eslint-disable-line react-hooks/exhaustive-deps

  useEffect(() => {
    buttonRef.current.addEventListener('onClick', onClick);
  }, []) // eslint-disable-line react-hooks/exhaustive-deps

  return (
    <>
      <lottus-button ref={buttonRef}></lottus-button>
    </>
  )
})


