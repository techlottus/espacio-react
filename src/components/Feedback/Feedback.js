import React, { useEffect } from 'react';
import "./Feedback.scss";

export const Feedback = ({data, onRight, text, html=() => {return}, isNoti=false}) => {

    const feedbackRef = React.createRef();
  
    useEffect(() => {
      feedbackRef.current.data = {
        left: data.left || {},
        right: data.right || {},
        type: data.type || '',
        isTextEvent: data.isTextEvent,
        textEvent: data.textEvent || '',
      };
      
    },[data,text]) // eslint-disable-line react-hooks/exhaustive-deps

    useEffect(() => {
      feedbackRef.current.addEventListener('onRight', onRight)
    },[]) // eslint-disable-line react-hooks/exhaustive-deps
    
    return (
      <>
        <lottus-feedback ref={feedbackRef}> 
          <div content="" className={`feedback-content ${isNoti ? "feedbackcontentmessagetext" : ""}`}>{text || html()}</div>
        </lottus-feedback>
      </>
    )
  }
