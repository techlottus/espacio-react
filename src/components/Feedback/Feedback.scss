@import "../../App.scss";

.feedback-content {
  font-family: "Nunito", sans-serif;
  line-height: pixeltorem(19px);
  font-size: pixeltorem(14px);

  @media only screen and (min-width: map-get($grid-breakpoints, m)) {
    font-size: pixeltorem(16px);
    line-height: pixeltorem(22px);
  }
}

.feedbackcontentmessagetext {
  overflow: hidden;
  text-overflow: ellipsis;
  display: -webkit-box;
  -webkit-line-clamp: 2; /* number of lines to show */
  line-clamp: 2;
  -webkit-box-orient: vertical;
}