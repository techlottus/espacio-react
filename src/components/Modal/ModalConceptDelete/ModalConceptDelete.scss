@import '../../../App.scss';
@import "../Modal.scss";

.modal-concepts {
  @extend .modal-container;
}
