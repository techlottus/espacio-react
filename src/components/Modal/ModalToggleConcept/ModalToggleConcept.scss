@import '../../../App.scss';
@import "../Modal.scss";

.modal-toggle {
  @extend .modal-container;
}
