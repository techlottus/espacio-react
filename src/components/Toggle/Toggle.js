import React, { useEffect } from 'react';



export const Toggle = React.memo(({data, onSwitch}) => {

    const toggleRef = React.createRef();
  
    useEffect(() => {
      toggleRef.current.data = {
        size: data.size ||'',
        status: data.status,
        disabled: data.disabled,
        test: data.test || '',
      };
    },[data]) // eslint-disable-line react-hooks/exhaustive-deps 

    useEffect(() => {
      toggleRef.current.addEventListener('onSwitch', onSwitch)
    },[]) // eslint-disable-line react-hooks/exhaustive-deps
  
    return (
      <>
        <lottus-toogle-switch ref={toggleRef}></lottus-toogle-switch>
      </>
    )
  });