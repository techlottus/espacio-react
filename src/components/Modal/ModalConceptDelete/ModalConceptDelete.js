import React, { useEffect, useRef, useState } from "react";
import { Button } from "../../Button/Button";
import {
  btnModalConceptCancel,
  btnModalConceptDelete,
} from "../../../constants/Payment.constant";
import "./ModalConceptDelete.scss";
import { useSelector } from "react-redux";

export const ModalConceptDelete = ({ isShow, onClose, handleDelete, outside }) => {

  const { texts } = useSelector(state => state.texts);
  const [isModal, setIsModal] = useState(false);

  const modalRef = useRef(); 

  useEffect(() => {
    setIsModal(isShow);
  }, [isShow]);

  useEffect(() => {
    if (isShow) {
      document.body.style.overflowY = "hidden";
    } else {
      document.body.style.overflowY = "auto";
    }
  }, [isShow]);

  useEffect(() => {
    const handleClickOutside = (event) => {
      if (
        modalRef.current &&
        !modalRef.current.contains(event.target) &&
        window.innerWidth > 991
      ) {
        outside();
      }
    };
    document.addEventListener("mousedown", handleClickOutside);
    return () => {
      document.removeEventListener("mousedown", handleClickOutside);
    };
  }, []); // eslint-disable-line react-hooks/exhaustive-deps

  const handleModal = () => {
    return (
      <div className="modal-concepts">
        <div className="modalcontent" ref={modalRef}>
          <div className="modalheader">
          <span className="material-icons icon iconright">{texts?.payment.modals.modalIcon}</span>
            <div className="title">
              {texts?.payment.modals.modalConceptDelete.title}
            </div>
          </div>
          <div className="modalbody">
            <div className="modalbtns">
              <div className="btnsecundary">
                <Button
                  data={{
                    ...btnModalConceptCancel,
                    title: texts?.payment.btnModal.cancel,
                    isExpand: window.innerWidth < 991,
                  }}
                  onClick={() => {
                    onClose();
                  }}
                />
              </div>
              <div className="btnprimary">
                <Button
                  data={{
                    ...btnModalConceptDelete,
                    title: texts?.payment.btnModal.delete,
                    isExpand: window.innerWidth < 991,
                  }}
                  onClick={() => {
                    handleDelete();
                  }}
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  };

  return <>{isModal ? handleModal() : ""}</>;
};
