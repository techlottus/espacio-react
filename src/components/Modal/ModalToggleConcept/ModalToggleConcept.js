import React, { useEffect, useRef, useState } from "react";
import { Button } from "../../Button/Button";
import {
  btnModalToggleAccept,
  btnModalToggleCancel,
} from "../../../constants/Payment.constant";
import "./ModalToggleConcept.scss";
import { useSelector } from "react-redux";
import { testPaymenTitle } from "../../../constants/test/testPayment";

export const ModalToggle = ({ isShow, onCancel, onNext, outside }) => {
  const { texts } = useSelector(state => state.texts);
  const [isModal, setIsModal] = useState(false);

  const modalRef = useRef();

  useEffect(() => {
    setIsModal(isShow);
  }, [isShow]);

  useEffect(() => {
    if (isShow) {
      document.body.style.overflowY = "hidden";
    } else {
      document.body.style.overflowY = "auto";
    }
  }, [isShow]);

  useEffect(() => {
    const handleClickOutside = (event) => {
      if (
        modalRef.current &&
        !modalRef.current.contains(event.target) &&
        window.innerWidth > 991
      ) {
        outside();
      }
    };
    document.addEventListener("mousedown", handleClickOutside);
    return () => {
      document.removeEventListener("mousedown", handleClickOutside);
    };
  }, []); // eslint-disable-line react-hooks/exhaustive-deps

  const handleModal = () => {
    return (
      <div className="modal-toggle">
        <div className="modalcontent" ref={modalRef}>
          <div className="modalheader">
            <span className="material-icons icon iconright">{texts?.payment.modals.modalIcon}</span>
            <div className="title">
              {texts?.payment.modals.modalToggleConcept.title}
            </div>
          </div>
          <div className="modalbody">
            <div className="modalbtns">
              <div className="btnsecundary">
                <Button
                  data={{
                    ...btnModalToggleCancel,
                    title: texts?.payment.btnModal.cancel,
                    isExpand: window.innerWidth < 991,
                    test: testPaymenTitle.btnModalToogleCancel
                  }}
                  onClick={() => {
                    onCancel();
                  }}
                />
              </div>
              <div className="btnprimary">
                <Button
                  data={{
                    ...btnModalToggleAccept,
                    title: texts?.payment.btnModal.accept,
                    isExpand: window.innerWidth < 991,
                    test: testPaymenTitle.btnModalToogleAccept
                  }}
                  onClick={() => {
                    onNext();
                  }}
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  };

  return <>{isModal ? handleModal() : ""}</>;
};
