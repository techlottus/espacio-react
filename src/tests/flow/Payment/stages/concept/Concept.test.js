import React from 'react';
import { mount } from 'enzyme';
import Concept from '../../../../../flow/Payment/stages/Concept/Concept';
import { Provider } from 'react-redux';
import '@testing-library/jest-dom';
import configureStore  from "redux-mock-store";
import thunk from "redux-thunk";
import { PaymentContext } from '../../../../../flow/Payment/PaymentContext';
import { paymentResponseConcepts } from '../../../../../fixtures/paymentFixture';
import { act } from 'react-dom/test-utils';
import { typesRequestErrors } from '../../../../../constants/error.constant';
import { Toggle } from '../../../../../components/Toggle/Toggle';
import { ModalToggle } from '../../../../../components/Modal/ModalToggleConcept/ModalToggleConcept';
import { ModalConceptDelete } from '../../../../../components/Modal/ModalConceptDelete/ModalConceptDelete';
import { Button } from '../../../../../components/Button/Button';
import { BrowserRouter } from 'react-router-dom';
import { Feedback } from '../../../../../components/Feedback/Feedback';
import { paymentText } from '../../../../../texts/paymentText';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const historyMock = { push: jest.fn(), location: {}, listen: jest.fn() };

window.dataLayer = {
  push: jest.fn()
}

let initState = {
  payment: {
    selectConcept: null,
    isInvoice: true,
    concepts: null,
    errors: {
      [typesRequestErrors.getConcepts]: {
        isError: false,
        msg: null
      },
      [typesRequestErrors.deleteConcept]: {
        isError: false,
        msg: null
      }
    }
  },
  texts: {
    payment: {
      ...paymentText
    }
  }
};

const handleSteps = jest.fn().mockImplementation(() => {});

let store = mockStore(initState);

const handleMount = (initState) => {
  store = mockStore(initState);
  return mount(
    <PaymentContext.Provider  value={{handleSteps}}>
      <Provider store={store}>
        <BrowserRouter>
          <Concept history={historyMock}  />
        </BrowserRouter>
      </Provider>
    </PaymentContext.Provider>
  )
}

describe('Test <Concept/>',() => {

  test('should init component',() => {
    const wrapper = handleMount(initState);
    expect(wrapper).toMatchSnapshot();
    wrapper.unmount();
  })

  describe('should load concepts',() => {
    test('with null',() => {
      initState = {
        ...initState,
        payment: {
          ...initState.payment,
          concepts: null
        }
      }
      const wrapper = handleMount(initState);
      expect(wrapper.find('CardDetail').exists()).toBe(false);
    })

    test('without null',() => {
      initState = {
        ...initState,
        payment: {
          ...initState.payment,
          concepts: [...paymentResponseConcepts]
        }
      }
      const wrapper = handleMount(initState);
      expect(wrapper.find('CardDeatil').exists()).toBe(true);
    })
  })

  describe('should run event ',() => {
    test('handleCheck',() => {
      initState = {
        ...initState,
        payment: {
          ...initState.payment,
          concepts: [...paymentResponseConcepts]
        }
      }
      const wrapper = handleMount(initState);
      act(() => {
        wrapper.find('CardDeatil').first().props().onCheck();
      })
      wrapper.update();
      expect( wrapper.find('CardDeatil').first().props().data.checked).toBe(true);
    })
  })

  test('should set concept default when back',() => {
    initState = {
      ...initState,
      payment: {
        ...initState.payment,
        selectConcept: {
          id: paymentResponseConcepts[0].transactionNumber
        }
      }
    }
    const wrapper = handleMount(initState);
    expect( wrapper.find('CardDeatil').first().props().data.checked).toBe(true);
  })

  describe('should action modal',() => {
    test('pass false to true',() => {
      initState = {
        ...initState,
        payment: {
          ...initState.payment,
        }
      }
      const wrapper = handleMount(initState);
      act(() => {
        wrapper.find(Toggle).props().onSwitch({
          detail: true
        });
      })
      wrapper.update();
      
    })
    test('pass true to false',() => {
      initState = {
        ...initState,
        payment: {
          ...initState.payment,
          isInvoice: true
        }
      }
      const wrapper = handleMount(initState);
      act(() => {
        wrapper.find(Toggle).props().onSwitch({
          detail: false
        });
      })
      wrapper.update();
      expect(wrapper.find(ModalToggle).props().isShow).toBe(true);
    })

    test('action back',() => {
      const wrapper = handleMount(initState);
      act(() => {
        wrapper.find(Toggle).props().onSwitch({
          detail: false
        });
        wrapper.find(ModalToggle).props().onCancel();
      });
      wrapper.update();
      expect(wrapper.find(Toggle).props().data.status).toBe(true);
      expect(wrapper.find(ModalToggle).props().isShow).toBe(false);
    })

    test('action next',() => {
      const wrapper = handleMount(initState);
      act(() => {
        wrapper.find(Toggle).props().onSwitch({
          detail: false
        });
        wrapper.find(ModalToggle).props().onNext();
      });
      wrapper.update();
      expect(wrapper.find(ModalToggle).props().isShow).toBe(false);
    })
  })

  test('should delete concept action close',() => {
    const wrapper = handleMount(initState);
    act(() => {
      wrapper.find('CardDeatil').first().props().onAction();
    })
    wrapper.update();
    expect(wrapper.find(ModalConceptDelete).props().isShow).toBe(true);
    act(() => {
      wrapper.find(ModalConceptDelete).props().onClose();
    })
    wrapper.update();
    expect(wrapper.find(ModalConceptDelete).props().isShow).toBe(false);
  })

  test('should delete concept action next',() => {
    const wrapper = handleMount(initState);
    act(() => {
      wrapper.find('CardDeatil').first().props().onAction();
    })
    wrapper.update();
    expect(wrapper.find(ModalConceptDelete).props().isShow).toBe(true);
    act(() => {
      wrapper.find(ModalConceptDelete).props().handleDelete();
    })
    wrapper.update();
    expect(wrapper.find(ModalConceptDelete).props().isShow).toBe(false);
  })

  describe('should action flow',() => {
    test('next with is invoice',() => {
      initState = {
        ...initState,
        payment: {
          ...initState.payment,
          selectConcept: {
            id: paymentResponseConcepts[0].transactionNumber
          },
          isInvoice: true
        }
      }
      const wrapper = handleMount(initState);
      act(() => {
        wrapper.find('.next').find(Button).props().onClick();
      })
      expect(historyMock.push.mock.calls[0][0]).toEqual('/payment/tax-data');
    })

    test('back with is not invoice',() => {
      initState = {
        ...initState,
        payment: {
          ...initState.payment,
          selectConcept: {
            id: paymentResponseConcepts[0].transactionNumber
          },
          isInvoice: false,
          errors:{
            [typesRequestErrors.deleteConcept]: {
              isError: true,
              msg: 'error en conceptos'
            },
          }
        }
      }
      const wrapper = handleMount(initState);
      act(() => {
        wrapper.find('.next').find(Button).props().onClick();
      })
      wrapper.update();
      expect(historyMock.push.mock.calls[0][0]).toEqual('/payment/methods');
    })

    test('close modal delete concept',() => {
      initState = {
        ...initState,
        payment: {
          ...initState.payment,
          selectConcept: {
            id: paymentResponseConcepts[0].transactionNumber
          },
          isInvoice: false,
          errors:{
            [typesRequestErrors.deleteConcept]: {
              isError: true,
              msg: 'error en conceptos'
            },
          }
        }
      }
      const wrapper = handleMount(initState);
      
    })
  })

})