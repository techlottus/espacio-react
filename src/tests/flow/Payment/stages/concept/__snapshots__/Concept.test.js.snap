// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <Concept/> should init component 1`] = `
Array [
  <div
    className="payment-concept"
  >
    <p
      className="paymentdescription"
    />
    <div
      className="paymentconcepts"
    />
    <div
      className="invoice"
    >
      <lottus-toogle-switch />
      <span />
    </div>
    <div
      className="next"
    >
      <lottus-button />
    </div>
  </div>,
  "",
]
`;
